edx/edx-enterprise
# ENT-289 Improve accounting for inactive courses

**Description:** We were missing a corner case where courses stop being returned for a catalog,
in which case we will need to mark them as inactive.

This change intoduces a lightweight audit summary of the course updates which
will be used in subsequent pushes to fully account for course state changes.

**JIRA:** https://openedx.atlassian.net/browse/ENT-289

**Dependencies:** n/a

**Merge deadline:** 4/7/2017

**Installation instructions:**
Make sure this version of edx-enterprise is installed in the environment.
Run the migrations for sap_success_factors.

Course Discovery
- At least two active courses
- At least one catalog containing at least one course

Enterprise customer
- Attached to catalog required above

Configured SapSuccessFactorsGlobalConfiguration (see document provided offline)
Configured SapSuccessFactorsEnterpriseCustomerConfiguration (see document provided offline)
Link to our existing EnterpriseCustomer

**Testing instructions:**

1. Run the following with the catalog having just one course (where `staff` is a local superuser with permissions to the course catalog):
    ```sh
    ./manage.py lms transmit_courseware_data --catalog_user staff --settings=devstack
    ```

2. Observe that in the output, that the active course was pushed to SuccessFactors. Check the sap_success_factors_catalogtransmissionaudit table to see that the audit summary reflects the course that was pushed and that there are no errors.

3. Add another active course to the catalog, re run the script, and observe that both courses are pushed in the same way as above.

4. Remove a course from the catalog, and run the script again. The removed course should show that the status for it was changed to INACTIVE. This can be found in the script output that is logged as well as in the sap_success_factors_catalogtransmissionaudit audit summary.

5. Run the script again, without changing the catalog. Observe that the audit summary for the latest push does not include the course that was removed in step 4.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:**
